### PR TITLE
EFS > IAM role for eks service account and efs helm deployment

### PR DESCRIPTION
**Pull request by janavenkat**
https://docs.aws.amazon.com/eks/latest/userguide/efs-csi.html

### CODE REVIEW

File: eks/iam.tf [hunk 1]
```diff
@@ -182,3 +182,58 @@ resource "aws_iam_role_policy_attachment" "fargate-AmazonEKSFargatePodExecutionR
...
+  version                       = "3.6.0"
+  create_role                   = true
+  role_name                     = "efs-driver"
+  provider_url                  = replace(aws_iam_openid_connect_provider.cluster.url, "https://", "")
+  role_policy_arns              = [aws_iam_policy.EFS-policy.arn]
+  # namespace and service account name
+  oidc_fully_qualified_subjects = ["system:serviceaccount:kube-system:efs-csi-controller-sa"]
+}
+
+resource "aws_iam_policy" "EFS-policy" {
```
Comment (abhiyerra) [inline eks/iam.tf:198]: Rename to "efs_policy"

File: eks/iam.tf [hunk 2]
```diff
@@ -182,3 +182,58 @@ resource "aws_iam_role_policy_attachment" "fargate-AmazonEKSFargatePodExecutionR
...
+  name        = "EFS-policy"
```
Comment (abhiyerra) [inline eks/iam.tf:199]: "${var.environment_name}-efs-policy"

File: eks/iam.tf [hunk 3]
```diff
@@ -182,3 +182,58 @@ resource "aws_iam_role_policy_attachment" "fargate-AmazonEKSFargatePodExecutionR
   policy_arn = "arn:aws:iam::aws:policy/AmazonEKSFargatePodExecutionRolePolicy"
   role       = aws_iam_role.fargate.name
 }
+
+module "iam_assumable_role_admin" {
+  count                         = var.efs_enabled ? 1 : 0
+  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
```
Comment (abhiyerra) [inline eks/iam.tf:188]: Where is this source? I don't think you added it.

File: eks/helm-efs.tf [hunk 4]
```diff
@@ -0,0 +1,41 @@
+//provider "helm" {
```
Comment (abhiyerra) [inline eks/helm-efs.tf:1]: Rename this file `efs.tf`

File: eks/helm-efs.tf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+//  count = var.efs_enabled ? 1 : 0
+//  kubernetes {
+//    config_path = "~/.kube/config"
+//  }
+//}
+//
```
Comment (abhiyerra) [inline eks/helm-efs.tf:7]: Remove this comments.

File: eks/helm-efs.tf [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  kubernetes {
+    host                   = aws_eks_cluster.cluster.endpoint
+    cluster_ca_certificate = base64decode(aws_eks_cluster.cluster.certificate_authority.0.data)
+    exec {
+      api_version = "client.authentication.k8s.io/v1alpha1"
+      args        = ["eks", "get-token", "--cluster-name", aws_eks_cluster.cluster.name]
+      command     = "aws"
+    }
+  }
+}
```
Comment (abhiyerra) [inline eks/helm-efs.tf:18]: This should provider block should be moved into `helm.tf`

File: eks/helm-efs.tf [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+resource "helm_release" "aws-efs-csi-driver" {
```
Comment (abhiyerra) [inline eks/helm-efs.tf:20]: Rename to "aws_efs_csi_driver"

**Review by abhiyerra — CHANGES_REQUESTED** [review]

File: eks/iam.tf (hunk 3 above)
Comment (janavenkat) [reply eks/iam.tf:188]: if you do `terraform init `from here it should work https://github.com/abhiyerra/Upwork/pull/3/files

Comment (janavenkat) [reply eks/iam.tf:188]: you mean to load the module from this repo itself?

**Review by abhiyerra — APPROVED** [review]

**Pull request merged**